### PR TITLE
Add more validation for ProxyConfig

### PR DESCRIPTION
**Pull request by ChenglongGao327**
add validation for ControlPlaneAuthPolicy/StatusPort

add test

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pkg/config/validation/validation.go [hunk 1]
```diff
@@ -1611,16 +1611,45 @@ func ValidateProxyConfig(config *meshconfig.ProxyConfig) (errs error) {
...
+		errs = multierror.Append(errs, multierror.Prefix(err, fmt.Sprintf("invalid status port:")))
 	}
 
 	return
 }
 
+// The default value is set to 189 characters.
+// Envoy's internal metrics take up 67 characters, for a total of 256 character name per metric.
+// Increase the value of this field if you find that the metrics from Envoys are truncated.
+func ValidateStatNameLength(len int32) error {
```
Comment (bianpengyuan) [inline pkg/config/validation/validation.go:1632]: Envoy has already removed `StatNameLength`. This option actually does not take any effect, so no need to add validation for it.

File: pkg/config/validation/validation.go [hunk 2]
```diff
@@ -1611,16 +1611,45 @@ func ValidateProxyConfig(config *meshconfig.ProxyConfig) (errs error) {
 		errs = multierror.Append(errs, multierror.Prefix(err, "invalid proxy admin port:"))
 	}
 
-	switch config.ControlPlaneAuthPolicy {
-	case meshconfig.AuthenticationPolicy_NONE, meshconfig.AuthenticationPolicy_MUTUAL_TLS:
-	default:
-		errs = multierror.Append(errs,
-			fmt.Errorf("unrecognized control plane auth policy %q", config.ControlPlaneAuthPolicy))
+	if err := ValidateControlPlaneAuthPolicy(config.ControlPlaneAuthPolicy); err != nil {
+		errs = multierror.Append(errs, multierror.Prefix(err, fmt.Sprintf("invalid authentication polic:")))
```
Comment (bianpengyuan) [inline pkg/config/validation/validation.go:1615]: nit: s/polic/policy

Comment (ChenglongGao327) [reply pkg/config/validation/validation.go]: done

File: pkg/config/validation/validation.go (hunk 1 above)
Comment (ChenglongGao327) [reply pkg/config/validation/validation.go]: > Envoy has already removed `StatNameLength`. This option actually does not take any effect, so no need to add validation for it.

ok,remove it

File: pkg/config/validation/validation_test.go [hunk 3]
```diff
@@ -167,6 +167,20 @@ func TestValidatePort(t *testing.T) {
 	}
 }
 
+func TestValidateControlPlaneAuthPolicy(t *testing.T) {
+	policys := map[meshconfig.AuthenticationPolicy]bool{
```
Comment (hzxuzhonghu) [inline pkg/config/validation/validation_test.go:171]: In go, table driven test is recommended 
https://github.com/golang/go/wiki/TableDrivenTests

Comment (ChenglongGao327) [reply pkg/config/validation/validation_test.go]: done

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**